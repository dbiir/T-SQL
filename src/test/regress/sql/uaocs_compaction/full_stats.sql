-- @Description Tests the behavior of full vacuum w.r.t. the pg_class statistics
-- ensure that the scan go through the index
CREATE TABLE uaocs_full_stats (a INT, b INT, c CHAR(128)) WITH (appendonly=true, orientation=column) DISTRIBUTED BY (a);
CREATE INDEX uaocs_full_stats_index ON uaocs_full_stats(b);
INSERT INTO uaocs_full_stats SELECT i as a, i as b, 'hello world' as c FROM generate_series(1, 50) AS i;
INSERT INTO uaocs_full_stats SELECT i as a, i as b, 'hello world' as c FROM generate_series(51, 100) AS i;
ANALYZE uaocs_full_stats;

SET enable_seqscan=false;
SELECT COUNT(*) FROM uaocs_full_stats;
SELECT relname, reltuples FROM pg_class WHERE relname = 'uaocs_full_stats';
SELECT relname, reltuples FROM pg_class WHERE relname = 'uaocs_full_stats_index';
DELETE FROM uaocs_full_stats WHERE a < 16;
SELECT COUNT(*) FROM uaocs_full_stats;
VACUUM FULL uaocs_full_stats;
SELECT COUNT(*) FROM uaocs_full_stats;
SELECT relname, reltuples FROM pg_class WHERE relname = 'uaocs_full_stats';
SELECT relname, reltuples FROM pg_class WHERE relname = 'uaocs_full_stats_index';
