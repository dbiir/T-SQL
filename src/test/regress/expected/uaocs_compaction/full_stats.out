-- @Description Tests the behavior of full vacuum w.r.t. the pg_class statistics
-- ensure that the scan go through the index
CREATE TABLE uaocs_full_stats (a INT, b INT, c CHAR(128)) WITH (appendonly=true, orientation=column) DISTRIBUTED BY (a);
CREATE INDEX uaocs_full_stats_index ON uaocs_full_stats(b);
INSERT INTO uaocs_full_stats SELECT i as a, i as b, 'hello world' as c FROM generate_series(1, 50) AS i;
INSERT INTO uaocs_full_stats SELECT i as a, i as b, 'hello world' as c FROM generate_series(51, 100) AS i;
ANALYZE uaocs_full_stats;
SET enable_seqscan=false;
SELECT COUNT(*) FROM uaocs_full_stats;
 count 
-------
   100
(1 row)

SELECT relname, reltuples FROM pg_class WHERE relname = 'uaocs_full_stats';
     relname      | reltuples 
------------------+-----------
 uaocs_full_stats |       100
(1 row)

SELECT relname, reltuples FROM pg_class WHERE relname = 'uaocs_full_stats_index';
        relname         | reltuples 
------------------------+-----------
 uaocs_full_stats_index |       100
(1 row)

DELETE FROM uaocs_full_stats WHERE a < 16;
SELECT COUNT(*) FROM uaocs_full_stats;
 count 
-------
    85
(1 row)

VACUUM FULL uaocs_full_stats;
SELECT COUNT(*) FROM uaocs_full_stats;
 count 
-------
    85
(1 row)

SELECT relname, reltuples FROM pg_class WHERE relname = 'uaocs_full_stats';
     relname      | reltuples 
------------------+-----------
 uaocs_full_stats |        85
(1 row)

SELECT relname, reltuples FROM pg_class WHERE relname = 'uaocs_full_stats_index';
        relname         | reltuples 
------------------------+-----------
 uaocs_full_stats_index |        85
(1 row)

